src/test/java/org/sagebionetworks/web/unitclient/presenter/EmailInvitationPresenterTest.java

```diff
@@ -0,0 +1,133 @@
+import org.sagebionetworks.web.client.GlobalApplicationState;
+import org.sagebionetworks.web.client.PlaceChanger;
+import org.sagebionetworks.web.client.SynapseFutureClient;
+import org.sagebionetworks.web.client.SynapseJavascriptClient;
+import org.sagebionetworks.web.client.place.EmailInvitation;
+import org.sagebionetworks.web.client.place.LoginPlace;
+import org.sagebionetworks.web.client.place.Profile;
+import org.sagebionetworks.web.client.presenter.EmailInvitationPresenter;
+import org.sagebionetworks.web.client.security.AuthenticationController;
+import org.sagebionetworks.web.client.view.EmailInvitationView;
+import org.sagebionetworks.web.client.view.users.RegisterWidget;
+import org.sagebionetworks.web.client.widget.entity.controller.SynapseAlert;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EmailInvitationPresenterTest {
+	@Mock private EmailInvitationView view;
+	@Mock private RegisterWidget registerWidget;
+	@Mock private SynapseJavascriptClient jsClient;
+	@Mock private SynapseFutureClient futureClient;
+	@Mock private SynapseAlert synapseAlert;
+	@Mock private AuthenticationController authController;
+	@Mock private GlobalApplicationState globalApplicationState;
+	@Mock private PlaceChanger placeChanger;
+	@Mock private EmailInvitation place;
+	@Mock private InviteeVerificationSignedToken inviteeVerificationSignedToken;
+	@Mock private MembershipInvtnSubmission mis;
+	@Mock private Team team;
+	@Mock private UserProfile inviterProfile;
+
+	private EmailInvitationPresenter presenter;
+	private String encodedMISignedToken;
+
+	@Before
+	public void before() {
+		when(globalApplicationState.getPlaceChanger()).thenReturn(placeChanger);
+		presenter = new EmailInvitationPresenter(
+				view, registerWidget, jsClient, futureClient, synapseAlert, authController, globalApplicationState);
+	}
+
+	public void beforeSetPlace(boolean loggedIn) {
+		when(authController.isLoggedIn()).thenReturn(loggedIn);
+		encodedMISignedToken = "encodedToken";
+		when(place.toToken()).thenReturn(encodedMISignedToken);
+		MembershipInvtnSignedToken decodedToken = new MembershipInvtnSignedToken();
+		when(futureClient.hexDecodeAndDeserialize(anyString(), eq(encodedMISignedToken))).thenReturn(getDoneFuture(decodedToken));
+		when(jsClient.getMembershipInvitation(decodedToken)).thenReturn(getDoneFuture(mis));
+		when(mis.getId()).thenReturn("misId");
+		when(mis.getInviteeEmail()).thenReturn("invitee@email.com");
+		when(mis.getTeamId()).thenReturn("teamId");
+		when(mis.getCreatedBy()).thenReturn("createdBy");
+		when(mis.getMessage()).thenReturn("message");
+	}
+
+	@Test
+	public void testLoggedIn() {
+		beforeSetPlace(true);
+		when(jsClient.getInviteeVerificationSignedToken(mis.getId())).thenReturn(getDoneFuture(inviteeVerificationSignedToken));
+		when(jsClient.updateInviteeId(inviteeVerificationSignedToken)).thenReturn(getDoneFuture(null));
+		presenter.setPlace(place);
+		verify(placeChanger).goTo(any(Profile.class));
```

Comment:
verify jsClient methods are called.  could capture Profile place as well, and verify it goes to the Teams area.